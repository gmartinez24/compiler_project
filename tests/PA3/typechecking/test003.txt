main {
    call println(5); // 1 type error

    call printInt(call readInt() + call readInt()); // fine

    call printInt(1 + call println(5+3) ); // 2 type errors 

}.