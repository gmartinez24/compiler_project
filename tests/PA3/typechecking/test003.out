Error type-checking file.
TypeError(2,5)[Call with args TypeList(int) matches no function signature.]
TypeError(6,23)[Call with args TypeList(int) matches no function signature.]
TypeError(6,21)[Cannot add int to ErrorType(Call with args TypeList(int) matches no function signature.).]
TypeError(6,5)[Call with args TypeList(ErrorType(Cannot add int to ErrorType(Call with args TypeList(int) matches no function signature.).)) matches no function signature.]

